src/storage/mod.rs

```diff
@@ -117,23 +117,29 @@ pub fn get_file_list<P: AsRef<Path>>(path: P) -> Result<Vec<PathBuf>, Error> {
     Ok(files)
 }
 
-pub(crate) enum Storage<'a> {
-    Database(DatabaseBackend<'a>),
-    S3(S3Backend<'a>),
+enum StorageBackend {
+    Database(DatabaseBackend),
+    S3(S3Backend),
 }
 
-impl<'a> Storage<'a> {
-    pub(crate) fn new(conn: &'a Connection) -> Self {
-        if let Some(c) = s3::s3_client() {
-            Storage::from(S3Backend::new(c, s3::S3_BUCKET_NAME))
+pub struct Storage {
+    backend: StorageBackend,
+}
+
+impl Storage {
+    pub fn new(pool: Pool) -> Self {
+        let backend = if let Some(c) = s3::s3_client() {
```

Comment:
Since this should now only be called by `Storage::new`, can we make `s3_client` `pub(super)`?

Reply:
It's also called in `crate::db::delete`. I plan to clean that up in a future PR.